lib/erb2epp/rewriter.rb

```diff
@@ -0,0 +1,215 @@
+# frozen_string_literal: true
+
+require 'ripper'
+
+module Erb2epp
+  # Rewrite code from Ruby ERB to Puppet EPP
+  class Rewriter
+    def initialize
+      @epp_params = []
+      @local_vars = []
```

Comment:
Should this be a set? That way you also handle if variables are later rewritten.

Thinking more on that: in Puppet it would probably be illegal so perhaps it should also be detected.

Reply:
Yes, set should be a better for this task, you're right!

Didn't get you regarding Puppet.. local variables can be used in EPP (in blocks e.g.).

Reply:
But can local variables change? Consider the following ERB:
```erb
<% x = 1 %>
<% x = 2 if @something%>
Something here <%= x %>
```

If you translate that to Puppet I think it'll be invalid because you can't assign `$x` again. Technically you should rewrite it to `$x = if @something { 1 } else { 2 }` but that's really hard to detect.

And writing this, you may also run into having to rewrite ternary expressions, or fail on them.

Reply:
Ah.. I'd say this is out of the scope of the current implementation.. What I'm doing here is just to collect the variable names and to prepend them with `$` later.